src/font.rs

```diff
@@ -107,12 +107,121 @@ impl<'a> Stream<'a> {
     }
 }
 
+/// The values of a [`font` shorthand](https://www.w3.org/TR/css-fonts-3/#font-prop).
+#[derive(Clone, PartialEq, Eq, Debug, Hash)]
+pub struct FontShorthand<'a> {
+    /// The font style.
+    pub font_style: Option<&'a str>,
+    /// The font variant.
+    pub font_variant: Option<&'a str>,
+    /// The font weight.
+    pub font_weight: Option<&'a str>,
+    /// The font stretch.
+    pub font_stretch: Option<&'a str>,
+    /// The font size.
+    pub font_size: &'a str,
+    /// The font family.
+    pub font_family: &'a str,
+}
+
+/// Split the font shorthand into its respective components.
+pub fn parse_font_shorthand(text: &str) -> Result<FontShorthand, Error> {
```

Comment:
Shouldn't we use `FromStr` here? Same with `FontFamily`.

Reply:
Yes, probably, will change.

Reply:
> Same with FontFamily.

For `FontFamily` it doesn't work because we are parsing a list of font families (i.e. it returns a vector of font families). We could create a wrapper struct around this vector, but I don't think that makes a lot of sense.

Reply:
I changed it for `FontShorthand`.

Reply:
What's wrong with returning a vector in FontFamily?